### PR TITLE
Add nf-core module and subworkflow `mirdeep2`

### PR DESCRIPTION
**Pull request by atrigila**
This PR adds modules and a subworkflow associated with `mirdeep2`.

Part of the migration from local to nf-core modules in https://github.com/nf-core/smrnaseq/issues/368 to match latest standards and improve quality.

Closes https://github.com/nf-core/smrnaseq/issues/431

## PR checklist

Closes #XXX 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

**Comment by atrigila** [comment]
There is an issue with `java.nio.channels.OverlappingFileLockException` in the conda subworkflow test

**Comment by atrigila** [comment]
The conda issue persists but, I will fix the new linting issues with the dev version of nf-core tools and `nf-core modules lint --fix`

File: modules/nf-core/mirdeep2/mirdeep2/main.nf [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def VERSION = '2.0.1'
+    """
+    miRDeep2.pl \\
+        $processed_reads \\
+        $fasta \\
+        $genome_mappings \\
+        $mature \\
+        none \\
```
Comment (sateeshperi) [inline modules/nf-core/mirdeep2/mirdeep2/main.nf:32]: what is none here? and missing $args

Comment (atrigila) [reply modules/nf-core/mirdeep2/mirdeep2/main.nf]: Thank you very much for spotting this! The software takes positional arguments and when these are not available they [are replaced by "none"](https://github.com/rajewsky-lab/mirdeep2?tab=readme-ov-file#example-use-2). I have now corrected this and improved the whole module :)

File: subworkflows/nf-core/fastq_find_mirna_mirdeep2/tests/main.nf.test.snap [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+    "smrnaseq - fasta - single_end": {
+        "content": [
+            [
+                "versions.yml:md5,10138b74aed5b2658c26ddf80ff391d5",
+                "versions.yml:md5,631c0428c28d5355f0e3e9bd790bd77d",
```
Comment (sateeshperi) [inline subworkflows/nf-core/fastq_find_mirna_mirdeep2/tests/main.nf.test.snap:6]: the workflow is emitting only version outputs ? no other outputs ?

Comment (atrigila) [reply subworkflows/nf-core/fastq_find_mirna_mirdeep2/tests/main.nf.test.snap:6]: captured the unstable outputs in the snapshot now :)

**Comment by atrigila** [comment]
Thank you @sateeshperi!

**Pull request merged**